server/DataAccessObjects/DriverDao.js

```diff
@@ -2,16 +2,80 @@ class DriverDao {
 
     constructor(gsDB) {
         this.gsDB = gsDB;
+        //this.driverDatabase = this.gsDB.collection("Drivers");
     }
 
     notifyAllValidDrivers(activeOrder) {
-        let driverDatabase = this.gsDB.collection("Drivers");
-        driverDatabase.where('capacity', '>=', activeOrder.getTotalQuantity()).get().then(snapshot => {
+        this.driverDatabase.where('capacity', '>=', activeOrder.getTotalQuantity()).get().then(snapshot => {
             snapshot.forEach(doc => {
                 activeOrder.notifyDriver(doc.data()["driverId"])
             });
         })
     }
+
+    createNewAccount(driver){
+        //check if driver Id doesn't exist
+        if (driver.getDriverId() == null){
+            var newId = this.generateUniqueKey();
+            driver.setDriverId(newId);
+        }
+
+        //this is required as new driver doesn't have any status yet and
+        //firebase doesn't allow data to be 'undefined'
+        if (driver.getDriverStatus() == null){
```

Comment:
also should be done in driver object constructor

Reply:
> also should be done in driver object constructor

I put this here as a sanity check because the case where driver status is an empty string is not covered in Driver constructor. This line of code is not necessary if we are always passing JSON object to Driver constructor where driver status is always defined.